### PR TITLE
Process reward modeling support

### PR DESCRIPTION
**Pull request by fabrahman**
This is a commit to support process reward modeling includes supports for:

1.  processing prm formatted data, 
2. computing process rewards at the end of each step, and
3. doing cross entropy over predicted logits over end of step tokens wrt labels.

Currently testing it here: https://beaker.org/ex/01J8AK58TPQ48SDH5AKSHVTZMK/tasks/01J8AK58TV4X7Y2VPQ9PG297WP/job/01J8AK5902D3935VH6SQMEMFV0

TODOs: refactoring the evaluation code for this prm task

### CODE REVIEW

**Comment by hamishivi** [comment]
@fabrahman do we still want to try to merge this? I think we have sort of moved on from PRMs...

**Comment by hamishivi** [comment]
Closing since we aren't focussing on PRMs rn.
